### PR TITLE
🌱 Update CAPD examples to use MachineDeployment

### PR DESCRIPTION
**Pull request by danniel1205**
**What this PR does / why we need it**:
This PR updates the existing two CAPD examples to use `MachineDeployment` instead of individual `Machine` and bumps the K8S version in those examples from v1.14.2 to v1.18.6. 

Using `MachineDeployment` is generally recommended as of today and there are also people asking for `MachineDeployment` examples in #cluster-api channel. So it would be nice we could have our existing examples up to date.

### CODE REVIEW

File: test/infrastructure/docker/examples/simple-cluster-without-kcp.yaml [hunk 1]
```diff
@@ -68,40 +68,51 @@ spec:
...
-  namespace: default
-spec:
-  joinConfiguration:
-    nodeRegistration:
-      kubeletExtraArgs:
-        eviction-hard: nodefs.available<0%,nodefs.inodesFree<0%,imagefs.available<0%
+  replicas: 1
+  selector:
+    matchLabels:
+      cluster.x-k8s.io/cluster-name: my-cluster
```
Comment (fabriziopandini) [inline test/infrastructure/docker/examples/simple-cluster-without-kcp.yaml:102]: nit: I think we can remove this label given it is automatically created by CAPI
(same in the template)
WDYT?

**Review by fabriziopandini — COMMENTED** [review]
@danniel1205 thanks for this PR!
overall lgtm, only one small nit

File: test/infrastructure/docker/examples/simple-cluster-without-kcp.yaml (hunk 1 above)
Comment (danniel1205) [reply test/infrastructure/docker/examples/simple-cluster-without-kcp.yaml:102]: @fabriziopandini yea, let me make the change.

Comment (danniel1205) [reply test/infrastructure/docker/examples/simple-cluster-without-kcp.yaml:100]: Looks like the label under `matchLabels` is required. I removed the redundant ones in `MachineDeployment.Metadata` and `MachineDeployment.Spec.Template`. Please help take a look. Thanks !
@fabriziopandini

**Review by fabriziopandini — COMMENTED** [review]
Thanks!
/approve

**Comment by fabriziopandini** [comment]
/lgtm
(just noticed the change was already approved)

**Pull request merged**